lib/errors/rescue_errors.rb

```diff
@@ -21,6 +29,26 @@ def self.rescue_unknown_format(base)
       end
     end # self.rescue_unknown_format
 
+    ##
+    # dynamically define each rescue in terms of the exception it will catch
+    AUTHENTICATION_ERRORS.each do |err|
+      send(:define_singleton_method, "rescue_#{err}") do |base|
+        base.rescue_from "Errors::AuthenticationErrors::#{err.to_s.camelcase}Error".constantize do |e|
```

Comment:
Let's not metaprogram these definitions, as we will want to have different messages for each one, and in some cases, the way we handle them may differ. Plus, we want to be able to search the codebase for the error name and find out where the handling is defined.